lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2308,7 +2314,9 @@ LLVMToSPIRVBase::transValueWithoutDecoration(Value *V, SPIRVBasicBlock *BB,
     // an OpenCL environment, we can use the same memory scope as OpenCL atomic
     // functions that don't have memory_scope argument i.e. memory_scope_device.
     // See the OpenCL C specification p6.13.11. "Atomic Functions"
-    Operands[1] = getUInt32(M, spv::ScopeDevice);
+    SmallVector<StringRef> SSIDs;
+    ARMW->getContext().getSyncScopeNames(SSIDs);
+    Operands[1] = getUInt32(M, map<Scope>(SSIDs[ARMW->getSyncScopeID()].str()));
```

Comment:
Do we need some error checks here? Invalid SyncScope ID or empty string etc?

Reply:
Empty string is a valid sync scope in LLVM, it's documented to be 'system' sync scope. Also there are no 'invalid' sync scopes. IMO they all should be defaulted to something, may be 'device'.

Reply:
Indeed, empty sting is mapped to 'device' scope now.